azure/services/managedclusters/spec.go

```diff
@@ -292,6 +302,17 @@ func buildAutoScalerProfile(autoScalerProfile *AutoScalerProfile) *asocontainers
 	return mcAutoScalerProfile
 }
 
+// getManagedClusterVersion gets the desired managed k8s version.
+// If autoupgrade channels is set to patch, stable or rapid, clusters can be upgraded to higher version by AKS.
```

Comment:
Do you think we can reuse the same function in `agentpools/spec.go`?

Reply:
Agentpools/spec.go requires asocontainerservicev1.ManagedClustersAgentPool as an input.